templates/bake/element/form.twig

```diff
@@ -3,7 +3,8 @@
     <fieldset>
         <legend><?= __('{{ action|humanize }} {{ singularHumanName }}') ?></legend>
         <?php
-{% for field in fields if field not in primaryKey %}
+{% for field in fields %}
+{%- if field not in primaryKey %}
```

Comment:
I don't think this extra `if` is necessary. Could just include the additional check in `if` statement below:

`{%- if keyFields[field] and field not in primaryKey %}`

Reply:
Yeah thats above my pay grade for twig knowledge. I did that and now its adding a control for id thats not expected by the tests

Reply:
Okay, I'll try to optimize it post merge.

Reply:
Ok yeah we want the extra if, combining them will add the primary key